lib/FocusTrap.js

```diff
@@ -1,31 +1,91 @@
 import PropTypes from 'prop-types';
 import React from 'react';
+import ReactDOM from 'react-dom';
+import {addFocusListener, addBlurListener} from './focusListeners';
 
 class FocusTrap extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.focusListener;
+    this.blurListener;
+    this.willWrap = !this.props.noWrapper;
+    this.setupNativeListeners = this.setupNativeListeners.bind(this);
+    this.removeNativeListeners = this.removeNativeListeners.bind(this);
+  }
+
   static propTypes = {
     onFocus: PropTypes.func,
     onBlur: PropTypes.func,
     component: PropTypes.any,
+    noWrapper: PropTypes.bool,
     children: PropTypes.node
   };
 
   static defaultProps = {
     component: 'div'
   };
 
+  componentDidMount() {
+    if (!this.willWrap) {
+      this.setupNativeListeners();
+    }
+  }
+
+  componentWillUnmount() {
+    this.removeNativeListeners();
+  }
+
+  setupNativeListeners() {
+    if (!this.focusListener) {
+      const {
+        onFocus, 
+        onBlur, 
+      } = this.props;
+
+      const elem = ReactDOM.findDOMNode(this);
+
+      if (elem && !this.focusListener) {
```

Comment:
Is `!this.focusListener` not redundant, given the outer condition already checks for this?

Reply:
Yep. Will clean up!